seacrowd/sea_datasets/asr_malcsc/asr_malcsc.py

```diff
@@ -0,0 +1,191 @@
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        data_paths = {
+            _DATASETNAME: Path(dl_manager.download_and_extract(_URLS[_DATASETNAME])),
+        }
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_paths[_DATASETNAME],
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        # read AUDIOINFO file
+        # columns: channel, uttrans_id, speaker_id, topic
+        audioinfo_filepath = os.path.join(filepath, "AUDIOINFO.txt")
+        with open(audioinfo_filepath, "r", encoding="utf-8") as audioinfo_file:
+            audioinfo_data = audioinfo_file.readlines()
+        audioinfo_data = audioinfo_data[1:]  # remove header
+        audioinfo_data = [s.strip("\n").split("\t") for s in audioinfo_data]
+
+        # read SPKINFO file
+        # columns: channel, speaker_id, gender, age, region, device
+        spkinfo_filepath = os.path.join(filepath, "SPKINFO.txt")
+        with open(spkinfo_filepath, "r", encoding="utf-8") as spkinfo_file:
+            spkinfo_data = spkinfo_file.readlines()
+        spkinfo_data = spkinfo_data[1:]  # remove header
+        spkinfo_data = [s.strip("\n").split("\t") for s in spkinfo_data]
+        for i, s in enumerate(spkinfo_data):
+            if s[2] == "M":
+                s[2] = "male"
+            elif s[2] == "F":
+                s[2] = "female"
+            else:
+                s[2] = None
+        # dictionary of metadata of each speaker
+        spkinfo_dict = {s[1]: {"speaker_gender": s[2], "speaker_age": int(s[3]), "speaker_region": s[4], "speaker_device": s[5]} for s in spkinfo_data}
+
+        num_sample = len(audioinfo_data)
+
+        for i in range(num_sample):
+            # wav file
+            wav_path = os.path.join(filepath, "WAV", audioinfo_data[i][1])
+            # transcription file
+            transcription_path = os.path.join(filepath, "TXT", audioinfo_data[i][1].replace("wav", "txt"))
+            with open(transcription_path, "r", encoding="utf-8") as transcription_file:
+                transcription = transcription_file.readlines()
+            # remove redundant speaker info from transcription file
+            transcription = [s.strip("\n").split("\t") for s in transcription]
+            transcription = [" ".join((s[0], s[-1])) for s in transcription]
+            text = " \n ".join(transcription)
```

Comment:
`text` field currently looks like this, I think the numbers in brackets are `[seconds_start, seconds_end]` or timeframe when text was said. I think this should be removed, but entries in this dataset are quite long and might benefit from the timestamps. What do you think @holylovenia @sabilmakbar @SamuelCahyawijaya?

![image](https://github.com/SEACrowd/seacrowd-datahub/assets/77622465/b575dc2a-1de2-46c0-918c-bede828327b4)

Reply:
Yes the numbers in brackets are the start and end times of the texts spoken. There are time gaps between the text snippets because the dataset contains conversations between 2 people, but the audio and transcript of each person is recorded in separate files. The timestamps can help to merge the conversation. Let me know how I should process this?

Reply:
Hi @zwenyu , the data loader works for me! So no issues with implementation!
As for the timestamps, I think we should include it in the text. Without it, can we still map the audio to the transcription? If not then maybe the timestamps are relevant 🤔 . Thoughts? @elyanah-aco

Reply:
Sorry, I missed this mention. What do you think about moving the timestamps to inside `metadata` instead?

cc: @zwenyu @ljvmiranda921 @elyanah-aco

Reply:
@holylovenia Can do that. Any preferred format? e.g.`[seconds_start, seconds_end]` corresponding to each text snippet on each row.

Reply:
I've added a `timestamp` field in the source schema in https://github.com/SEACrowd/seacrowd-datahub/pull/494/commits/8dc16cbeb89a6141bb8ff0f2b509e0e7f6b70faf. The seacrowd sptext schema does not have `timestamp` in `metadata` so I did not add it there.

Reply:
@zwenyu Could you please add `timestamp` under `metadata` in the `seacrowd_sptext` schema as well? It should be fine as long as the unit test doesn't raise an error.

Reply:
Added.